src/pages/workspaces/workspace/DeleteWorkspaceModal.js

```diff
@@ -60,28 +73,35 @@ const DeleteWorkspaceModal = ({ workspace: { workspace: { namespace, name, bucke
     ])
   }
 
-  const getAppDeletionMessage = () => {
-    return !_.isEmpty(nonDeletableApps) ?
+  const hasApps = () => {
+    return deletableApps !== undefined && nonDeletableApps !== undefined &&
+      (!_.isEmpty(deletableApps) ||
+        !_.isEmpty(nonDeletableApps))
+  }
+
+  const isDeleteDisabledFromResources = (hasApps() && !_.isEmpty(nonDeletableApps)) || controlledResourcesExist
```

Comment:
I was thinking of "resource" as a more generic term that could be either an app or a controlled resource.